contracts/policy-agents/PolicyOfferExecutor.sol

```diff
@@ -0,0 +1,129 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../interfaces/IClaimsManager.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract PolicyOfferExecutor {
+    using ECDSA for bytes32;
+    using SafeERC20 for IERC20;
+
+    address public immutable claimsManager;
+    address public immutable offerSigner;
+    address public immutable token;
+    address public immutable beneficiary;
```

Comment:
Just some thoughts on these being immutable 🤔 
1. What if we want to receive payments in ETH or any other ERC20 token? Maybe `token` should instead be an array of supported tokens and `executeOffer()` will need to have a token argument as well.
2. Who would `offerSigner` and `beneficiary` be? For instance if we already know that beneficiary will be the DAO treasury or similar then we might want to just hard code it here instead of setting it in the constructor, right? if `offerSigner` is a multisig then we could also hard code it and put a comment listing the verified signers to provide more transparency.

Reply:
1. I think the better approach here is for there to be an `offerToken` that is signed by the offer signer along with `offerAmount`. However, I'm pretty sure that we won't use anything other than USDC due to API provider requirements.
2. `offerSigner` is an EOA whose mnemonic we keep in our backend, `beneficiary` is the manager multisig. Declaring a variable to be immutable and hardcoding it as a constants in the contract are equivalent, this is just easier to work with for testing and such.

Reply:
I suppose a downside to 1 is that the user now has to also check the address of the token for which the offer is, otherwise we can have them execute an offer for a more valuable token